File: JavaScript/terminal_games/number_guessing_game.js
import { createInterface } from "readline";

const rl = createInterface({
    input: process.stdin,
    output: process.stdout,
});

const num = Math.floor(Math.random() * 101);
let tries = 1;

console.log("Number Guessing Game!");
console.log("*********************\n");

function askQuestion() {
    rl.question("Enter a number between 1 to 100: ", (answer) => {
        const guess = parseInt(answer);

        if (isNaN(guess) || guess < 1 || guess > 100) {
            console.log("Please enter a valid number between 1 and 100.");
            askQuestion();
        } else if (guess < num) {
            console.log("Too low! Try again.");
            tries++;
            askQuestion();
        } else if (guess > num) {
            console.log("Too high! Try again.");
            tries++;
            askQuestion();
        } else {
            console.log(
                `Congratulations! You guessed the number ${num} in ${tries} tries.`
            );
            rl.close();
        }
    });
}

askQuestion();

File: Lua/terminal_games/number_guessing_game.lua
math.randomseed(os.time() + math.floor(os.clock() * 1000))
math.random(); math.random(); math.random()

print("Number Guessing Game!")
print(("*"):rep(21))

local n = math.random(1, 100)
local tries = 0

while true do
    io.write("Guess a number between 1 and 100: ")
    local line = io.read()

    local guess = tonumber(line)

    if guess == nil then
        print("Please enter a valid number.")
    elseif guess < n then
        print("Too low! Try again.\n")
        tries = tries + 1
    elseif guess > n then
        print("Too high! Try again.\n")
        tries = tries + 1
    else
        print("\nCongratulations! You guessed the number!")
        print("It took you " .. tries .. " tries.")
        break
    end
end
